File: 0198-house-robber/0198-house-robber.ts
function rob(nums: number[]): number {
    let prev2 = 0, prev1 = 0;
    for (let i = 0; i < nums.length; i++) {
        const cur = Math.max(nums[i] + prev2, prev1);
        prev2 = prev1;
        prev1 = cur;
    }
    return prev1;
}

File: 0198-house-robber/0198-house-robber.py
class Solution(object):
    def rob(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        prev1,prev2=0,0

        for num in nums:
            current = max(prev1,prev2+num)
            prev2 = prev1
            prev1= current
        return prev1